recipes-kernel/linux/files/openxt-kmeta/bsp/common-pc/common-pc-drivers-reduced.cfg

```diff
@@ -0,0 +1,114 @@
+#
+# Serial ATA and Parallel ATA drivers (libata)
+#
+CONFIG_ATA=y
+CONFIG_ATA_ACPI=y
+CONFIG_SATA_AHCI=y
+
+#
+# Input device support
+#
+CONFIG_INPUT=y
+CONFIG_INPUT_LEDS=m
+
+#
+# Userland interfaces
+#
+CONFIG_INPUT_MOUSEDEV=y
+CONFIG_INPUT_EVDEV=y
+
+#
+# Input Device Drivers
+#
+CONFIG_INPUT_KEYBOARD=y
+CONFIG_KEYBOARD_ATKBD=y
+CONFIG_INPUT_MOUSE=y
+
+CONFIG_MOUSE_PS2=m
+CONFIG_MOUSE_PS2_ALPS=y
+CONFIG_MOUSE_PS2_BYD=y
+CONFIG_MOUSE_PS2_LOGIPS2PP=y
+CONFIG_MOUSE_PS2_SYNAPTICS=y
+CONFIG_MOUSE_PS2_SYNAPTICS_SMBUS=y
+CONFIG_MOUSE_PS2_CYPRESS=y
+CONFIG_MOUSE_PS2_LIFEBOOK=y
+CONFIG_MOUSE_PS2_TRACKPOINT=y
+CONFIG_MOUSE_PS2_FOCALTECH=y
+
+CONFIG_MOUSE_APPLETOUCH=m
+CONFIG_MOUSE_BCM5974=m
+CONFIG_MOUSE_SYNAPTICS_I2C=m
+CONFIG_INPUT_TABLET=y
+
+CONFIG_INPUT_TOUCHSCREEN=y
+CONFIG_TOUCHSCREEN_PROPERTIES=y
+CONFIG_TOUCHSCREEN_ATMEL_MXT=m
+CONFIG_TOUCHSCREEN_WACOM_W8001=m
+CONFIG_TOUCHSCREEN_WACOM_I2C=m
+CONFIG_TOUCHSCREEN_USB_COMPOSITE=m
+CONFIG_TOUCHSCREEN_USB_EGALAX=y
+CONFIG_TOUCHSCREEN_USB_PANJIT=y
+CONFIG_TOUCHSCREEN_USB_3M=y
+CONFIG_TOUCHSCREEN_USB_ITM=y
+CONFIG_TOUCHSCREEN_USB_ETURBO=y
+CONFIG_TOUCHSCREEN_USB_GUNZE=y
+CONFIG_TOUCHSCREEN_USB_DMC_TSC10=y
+CONFIG_TOUCHSCREEN_USB_IRTOUCH=y
+CONFIG_TOUCHSCREEN_USB_IDEALTEK=y
+CONFIG_TOUCHSCREEN_USB_GENERAL_TOUCH=y
+CONFIG_TOUCHSCREEN_USB_GOTOP=y
+CONFIG_TOUCHSCREEN_USB_JASTEC=y
+CONFIG_TOUCHSCREEN_USB_ELO=y
+CONFIG_TOUCHSCREEN_USB_E2I=y
+CONFIG_TOUCHSCREEN_USB_ZYTRONIC=y
+CONFIG_TOUCHSCREEN_USB_ETT_TC45USB=y
+CONFIG_TOUCHSCREEN_USB_NEXIO=y
+CONFIG_TOUCHSCREEN_USB_EASYTOUCH=y
+
+CONFIG_INPUT_MISC=y
+CONFIG_INPUT_UINPUT=m
+
+#
+# Hardware I/O ports
+#
+CONFIG_SERIO=y
+CONFIG_SERIO_I8042=y
+CONFIG_SERIO_SERPORT=y
+
+CONFIG_RTC_CLASS=y
+CONFIG_RTC_DRV_CMOS=y
+
+#
+# Character devices
+#
+CONFIG_TTY=y
+CONFIG_VT=y
+
+#
+# Console display support
+#
+CONFIG_VGA_CONSOLE=y
+
+#
+# I2C support
+#
+CONFIG_I2C=y
```

Comment:
I've only skimmed.  This is better than the full defconfigs we currently use, but I am a little surprised by how how many lines are required for all the fragments.  Picking I2C as an example, this seems like something we shouldn't need to enable explicit.  Looks like, CONFIG_I2C_I801=y would select I2C_SMBUS... I think that would indirectly turn I2C on?

I'm guessing you started from a full defconfig?  I recently found out about `make savedefconfig`.  That saves only the non-default values, so the full .config is created after running through `make oldconfig`.  For NDVM, `make savedefconfig` creates a 200-something line defconfig.  Making fragments out of that should be much smaller.

Reply:
Actually I used `bitbake -c diffconfig` quite a bit to try and get as close as possible to the original defconfigs in order to not introduce too much significant changes.

That specific line comes directly from [the original file in yocto-kernel-cache](https://git.yoctoproject.org/cgit/cgit.cgi/yocto-kernel-cache/tree/bsp/common-pc/common-pc-drivers.cfg).

I agree though, that would avoid keeping some configuration enabled should another configuration fragment disable one that would have selected it.